mkosi/config.py

```diff
@@ -5235,19 +5326,28 @@ def parse_config(
             context.parse_config_one(configdir, parse_profiles=True, parse_local=True)
 
     config = context.finalize()
+    main = Config.from_dict(config)
 
-    if config["history"] and want_new_history(args):
+    if main.history and want_new_history(args):
         Path(".mkosi-private/history").mkdir(parents=True, exist_ok=True)
         Path(".mkosi-private/history/latest.json").write_text(dump_json(Config.to_partial_dict(cli)))
 
     tools = None
-    if config.get("tools_tree") == Path("default"):
+    if main.tools_tree == Path("default"):
         if in_box():
             config["tools_tree"] = Path(os.environ["MKOSI_DEFAULT_TOOLS_TREE_PATH"])
         else:
             tools = finalize_default_tools(context, config, configdir=configdir, resources=resources)
             config["tools_tree"] = tools.output_dir_or_cwd() / tools.output
 
+    if want_default_initrd(main):
+        initrd = finalize_default_initrd(context, config, resources=resources)
+        config["initrds"] = [
+            initrd.output_dir_or_cwd() / initrd.output if i == Path("default") else i for i in main.initrds
```

Comment:
Since this line lost some indentation now a linebreak got removed, let's add some parentheses so it's less glyph soup.
```suggestion
            (initrd.output_dir_or_cwd() / initrd.output if i == Path("default") else i) for i in main.initrds
```